pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1739,8 +1739,10 @@ class Timedelta(_Timedelta):
                 + int(kwargs.get("milliseconds", 0) * 1_000_000)
                 + seconds
             )
-
-        if unit in {"Y", "y", "M"}:
+        if unit in {"Y", "y", "M"} or (isinstance(value, np.timedelta64) and
```

Comment:
I think it's still better to handle this in the `is_timedelta64_object` branch

Reply:
I can move this back into `is_timedelta64_object` if we're ok with the `NPY_DATETIMEUNIT` checks. 
By that time the `value` has been converted into a numpy object so the `is_support_unit` check always passes

Reply:
I think you can do:

```
reso = get_datetime64_unit(value)
if not (is_supported_unit(reso) or reso == NPY_DATETIMEUNIT.NPY_FR_D):
    raise ValueError(...)
```

Reply:
Just sanity checked again and unless I'm misunderstanding, your solution doesn't work. 

**Adding:** 
```python
if not (is_supported_unit(reso) or reso == NPY_DATETIMEUNIT.NPY_FR_D):
      err = npy_unit_to_abbrev(reso)
     raise ValueError(f" cannot construct a Timedelta from a unit {err}")
```

**Causes `pd.Timedelta(days=3)` to raise:** 
```python
File "timedeltas.pyx", line 1796, in pandas._libs.tslibs.timedeltas.Timedelta.__new__
ValueError:  cannot construct a Timedelta from a unit ns
```


**Which seems wrong given `is_supported_unit`:**
```python
def is_supported_unit(NPY_DATETIMEUNIT reso):
    return (
        reso == NPY_DATETIMEUNIT.NPY_FR_ns
        or reso == NPY_DATETIMEUNIT.NPY_FR_us
        or reso == NPY_DATETIMEUNIT.NPY_FR_ms
        or reso == NPY_DATETIMEUNIT.NPY_FR_s
    )
```

Reply:
You might need to allow `NPY_FR_GENERIC` as well

Reply:
In my example above for `pd.Timedelta(days=3)`

the `value` going into `get_datetime64_unit(value)` is a `timedelta64` but I think it expects a `datetime64` and might be what's causing the issue.

Reply:
Yah `NPY_FR_GENERIC` was needed. 

Hitting other issues now after running a full `pytest`. 

Things like: `pd.Timedelta(np.timedelta64(1, m))` or pd.Timedelta(np.timedelta64(1, h))

I could solve this with by adding a `is_ambiguous()` in  dtypes?

Reply:
Ah I guess minute and hour should also be accepted in the `if` condition then too